src/algebra/algebra/basic.lean

```diff
@@ -518,6 +518,12 @@ lemma map_finsupp_sum {α : Type*} [has_zero α] {ι : Type*} (f : ι →₀ α)
 @[simp] lemma map_bit1 (x) : φ (bit1 x) = bit1 (φ x) :=
 φ.to_ring_hom.map_bit1 x
 
+-- TODO[gh-6025]: make this an instance once safe to do so
+def subsingleton {R A B} [comm_semiring R] [semiring A] [semiring B]
+  [algebra R A] [algebra R B] [subsingleton A] :
```

Comment:
I think the typeclasses are already in this section. I also worry that your current statement may have the wrong universe variables.
```suggestion
lemma subsingleton [subsingleton A] :
```